app/scripts/modules/core/utils/timeFormatters.ts

```diff
@@ -0,0 +1,84 @@
+import { IComponentController, IComponentOptions, module } from 'angular';
+import * as moment from 'moment';
+
+import { SETTINGS } from 'core/config/settings';
+
+export function duration(input: any) {
+  if (!input || isNaN(input) || input < 0) {
+    return '-';
+  }
+  const thisMoment = moment.utc(isNaN(parseInt(input, 10)) ? input : parseInt(input, 10));
+  const format = thisMoment.hours() ? 'HH:mm:ss' : 'mm:ss';
+  let dayLabel = '';
+  if (thisMoment.isValid()) {
+    const days = Math.floor(input / 86400000);
```

Comment:
Can you add a comment about what 86400000 represents? (hint, it's a full day, I just did some maths)